tests/tests/debug_api.rs

```diff
@@ -44,117 +44,34 @@ async fn test_raw_transaction(#[future] katana: Katana, _setup: ()) {
-    let transaction = TransactionSigned::decode_enveloped(&mut rlp_bytes.unwrap().as_ref()).unwrap();
-    let signer = transaction.recover_signer().unwrap();
     let transaction = from_recovered_with_block_context(
         TransactionSignedEcRecovered::from_signed_transaction(transaction, signer),
-        *BLOCK_HASH,
-        BLOCK_NUMBER,
-        None,
-        0,
+        tx.block_hash.unwrap(),
+        tx.block_number.unwrap(),
+        header.base_fee_per_gas.map(|x| x.try_into().unwrap()),
+        tx.transaction_index.unwrap() as usize,
     );
-    let res = reqwest_client
-        .post(format!("http://localhost:{}", server_addr.port()))
-        .header("Content-Type", "application/json")
-        .body(
-            RawRpcParamsBuilder::new("eth_getTransactionByHash")
-                .add_param(format!("0x{:064x}", *EIP2930_TX_HASH))
-                .build(),
-        )
-        .send()
-        .await
-        .expect("Failed to call Debug RPC");
-    let response = res.text().await.expect("Failed to get response body");
-    let response: Value = serde_json::from_str(&response).expect("Failed to deserialize response body");
-    let rpc_transaction: reth_rpc_types::Transaction =
-        serde_json::from_value(response["result"].clone()).expect("Failed to deserialize result");
-    assert_eq!(transaction, rpc_transaction);
 
-    // Legacy
-    let reqwest_client = reqwest::Client::new();
-    let res = reqwest_client
-        .post(format!("http://localhost:{}", server_addr.port()))
-        .header("Content-Type", "application/json")
-        .body(
-            RawRpcParamsBuilder::new("debug_getRawTransaction")
-                .add_param(format!("0x{:064x}", *LEGACY_TX_HASH))
-                .build(),
-        )
-        .send()
-        .await
-        .expect("Failed to call Debug RPC");
-    let response = res.text().await.expect("Failed to get response body");
-    let raw: Value = serde_json::from_str(&response).expect("Failed to deserialize response body");
-    let rlp_bytes: Option<Bytes> = serde_json::from_value(raw["result"].clone()).expect("Failed to deserialize result");
-    assert!(rlp_bytes.is_some());
-    // We can decode the RLP bytes to get the transaction and compare it with the original transaction
-    let transaction = TransactionSigned::decode_enveloped(&mut rlp_bytes.unwrap().as_ref()).unwrap();
-    let signer = transaction.recover_signer().unwrap();
-    let transaction = from_recovered_with_block_context(
-        TransactionSignedEcRecovered::from_signed_transaction(transaction, signer),
-        *BLOCK_HASH,
-        BLOCK_NUMBER,
-        None,
-        0,
-    );
+    // Checking some fields but not all since mocked signer is not exact at the moment
+    assert_eq!(transaction.block_hash, header.hash);
+    assert_eq!(transaction.block_number, header.number);
+    assert_eq!(transaction.nonce, tx.nonce);
+    assert_eq!(transaction.chain_id, tx.chain_id);
```

Comment:
@greged93 Here is the only place in the PR where we slightly lose information but I think that with what I did here it is equivalent:
- As we now use randomly generated transactions, we no longer need to do a test per transaction type, we are quite covered with a single call (moreover, the conversion with the types is handled by reth so we do not need to do as many tests).
- Where we are slightly losing is that the signing of the tx is not correct as you pointed out in your comments here https://github.com/kkrt-labs/kakarot-rpc/pull/1324#issuecomment-2278814099, so we can no longer use the approach aimed at reconstructing the transaction with the Bytes to compare it to the one obtained by get transaction by hash because some elements such as the hash for example which is recalculated do not match. To do this I adopted the following approach:
    - I made the first call with `debug_getRawTransaction`, then I rebuilt the transaction despite the fact that the signer was not the right one.
    - I then compare a few fields that I know are correct with my Katana transaction that I have in memory.
    - I redo my second test with `eth_getTransactionByHash` and this time I compare my full result with my Katana transaction that I have in memory.
    - So I think we are sufficiently covered here, especially since the conversion parts are handled by reth so we should not need to test everything.

I did all this to try to simplify our test environment as much as possible and remove all the constants and heavy functions that were only there for this test. Tell me what you think but I think the coverage remains largely reasonable here. And if we are not 100% happy, in follow up, we can always improve our mocking to try to have an exact transaction with the right signer.

Reply:
@greged93 Now this should be good, I used the reth test utils `sign_tx_with_random_key_pair` method to sign the random transaction in the arbitrary generation. I will fix the `arbitrary_with_optional_fields` to `arbitrary` in a follow up PR as there are several places to change in the code and it is not totally related to this PR which is already pretty big to review :)